src/FffGcodeWriter.cpp

```diff
@@ -1460,66 +1466,105 @@ bool FffGcodeWriter::addSupportRoofsToGCode(const SliceDataStorage& storage, Lay
+
+    const coord_t support_bottom_line_distance = bottom_extr.getSettingInMicrons("support_bottom_line_distance");
+    Infill bottom_computation(pattern, support_layer.support_bottom, outline_offset, gcode_layer.configs_storage.support_bottom_config.getLineWidth(), support_bottom_line_distance, support_bottom_overlap, fill_angle, z, extra_infill_shift, perimeter_gaps, connected_zigzags, use_endpieces);
+    Polygons bottom_polygons;
+    Polygons bottom_lines;
+    bottom_computation.generate(bottom_polygons, bottom_lines);
+    if (bottom_polygons.empty() && bottom_lines.empty())
     {
-        fillAngle = 90; // perpendicular to support lines
+        return false;
     }
-    else 
+    setExtruder_addPrime(storage, gcode_layer, layer_nr, bottom_extruder_nr);
+    gcode_layer.addPolygonsByOptimizer(bottom_polygons, &gcode_layer.configs_storage.support_bottom_config);
+    gcode_layer.addLinesByOptimizer(bottom_lines, &gcode_layer.configs_storage.support_bottom_config, (pattern == EFillMethod::ZIG_ZAG) ? SpaceFillType::PolyLines : SpaceFillType::Lines);
+    return true;
+}
+
+double FffGcodeWriter::supportInterfaceFillAngle(const SliceDataStorage& storage, const EFillMethod pattern, const int layer_number) const
+{
+    if (pattern == EFillMethod::CONCENTRIC)
     {
-        fillAngle = 45 + (((layer_nr % 2) + 2) % 2) * 90; // alternate between the two kinds of diagonal:  / and \ .
-        // +2) %2 to handle negative layer numbers
+        return 0; //Concentric has no rotation.
+    }
+    if (pattern == EFillMethod::TRIANGLES)
+    {
+        return 90; //Triangular support interface shouldn't alternate every layer.
     }
-    int support_skin_overlap = 0; // the skin (roofs/bottoms) should never be expanded outwards
-    int outline_offset =  0;
-    int extra_infill_shift = 0;
-    Polygons* perimeter_gaps = nullptr;
-    bool use_endpieces = true;
-    bool connected_zigzags = false;
-    Infill infill_comp(pattern, support_layer.skin, outline_offset, gcode_layer.configs_storage.support_interface_config.getLineWidth(), support_line_distance, support_skin_overlap, fillAngle, z, extra_infill_shift, perimeter_gaps, connected_zigzags, use_endpieces);
-    Polygons support_polygons;
-    Polygons support_lines;
-    infill_comp.generate(support_polygons, support_lines);
 
-    if (support_lines.size() > 0 || support_polygons.size() > 0)
+    for (const SliceMeshStorage& mesh : storage.meshes)
     {
-        setExtruder_addPrime(storage, gcode_layer, layer_nr, skin_extruder_nr);
-        gcode_layer.addPolygonsByOptimizer(support_polygons, &gcode_layer.configs_storage.support_interface_config);
-        gcode_layer.addLinesByOptimizer(support_lines, &gcode_layer.configs_storage.support_interface_config, (pattern == EFillMethod::ZIG_ZAG)? SpaceFillType::PolyLines : SpaceFillType::Lines);
-        added = true;
+        if (mesh.getSettingInMicrons("support_roof_height") >= 2 * getSettingInMicrons("layer_height"))
```

Comment:
This was always the case: https://github.com/Ultimaker/CuraEngine/blame/master/src/FffGcodeWriter.cpp#L1535